src/ejabberd_cowboy.erl

```diff
@@ -127,12 +128,13 @@ do_start_cowboy(Ref, Opts, Retries, SleepTime) ->
 do_start_cowboy(Ref, Opts) ->
     SSLOpts = gen_mod:get_opt(ssl, Opts, undefined),
     NumAcceptors = gen_mod:get_opt(num_acceptors, Opts, 100),
-    TransportOpts = gen_mod:get_opt(transport_options, Opts, []),
+    TransportOpts0 = gen_mod:get_opt(transport_options, Opts, #{}),
+    TransportOpts = TransportOpts0#{numb_acceptros => NumAcceptors},
     Modules = gen_mod:get_opt(modules, Opts),
     Dispatch = cowboy_router:compile(get_routes(Modules)),
     ProtocolOpts = [{env, [{dispatch, Dispatch}]} |
                     gen_mod:get_opt(protocol_options, Opts, [])],
-    case catch start_http_or_https(SSLOpts, Ref, NumAcceptors, TransportOpts, ProtocolOpts) of
+    case catch start_http_or_https(SSLOpts, Ref, TransportOpts, ProtocolOpts) of
```

Comment:
why `case catch ... `?
how about using
`try ... of ... catch...`??

Reply:
You are right, `try .. of ... catch ...` would be better. I just didn't want to change it as it was not the subject of the PR. The change is rather easy so I can do it in this PR anyway. What do you think?

Reply:
No pressure for this change PR single responsibility 👍